rust/common/error/macros.rs

```diff
@@ -24,117 +24,86 @@
 macro_rules! error_messages {
     {
         $name:ident code: $code_pfx:literal, type: $message_pfx:literal,
-        $(
-            $error_name:ident( $($inner:ty),* $(,)? ) = $code:literal: $body:literal
-        ),* $(,)?
+        $($error_name:ident( $($inner:ty),* $(,)? ) = $code:literal: $body:literal),+ $(,)?
     } => {
-        error_messages_impl!(
-            $name, $code_pfx, max!($($code),*), $message_pfx,
-            $(($error_name, $code, $body, ($($inner),*))),*
-        );
-    };
-}
-
-macro_rules! error_messages_impl {
-    (
-        $name:ident, $code_pfx:literal, $max_code:expr, $message_pfx:literal,
-        $(($error_name:ident, $code:literal, $body:literal, ($($inner:ty),*))),*
-    ) => {
         #[derive(Clone, Debug, Eq, PartialEq)]
         pub enum $name {$(
             $error_name($($inner),*),
         )*}
 
         impl $name {
-            pub fn code(&self) -> usize {
-                match self {$(
-                    Self::$error_name(..) => $code,
-                )*}
+            const fn __max_code() -> usize {
```

Comment:
let's use just the most standard rust until we converge on a Rust code style for the company